src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -48,29 +47,26 @@ internal class SNITCPHandle : SNIHandle
         /// </summary>
         public override void Dispose()
         {
-            lock (this)
+            if (_sslOverTdsStream != null)
```

Comment:
Does the lock removal provide any perf benefits? If yes, can you provide a measurement?

Reply:
It seems there is slight perf improvement without locks.
I ran benchmark test sync / async, with / without lock 5 times each.

Here is the result:
sync with lock
12115 tps
11979 tps
11720 tps
12141 tps
12317 tps

sync without lock
12786 tps
12454 tps
12552 tps
12486 tps
12344 tps

async with lock
12616 tps
12318 tps
12475 tps
12040 tps
12617 tps

async without lock
12620 tps
12591 tps
12551 tps
12233 tps
12643 tps

Reply:
What was the purpose of the lock?  I like seeing pointless locks removed, as locks incur both overhead even for single-threaded usage as well as serve as a point of contention and scalability bottleneck.  But was it pointless?